src/main/java/io/confluent/idesidecar/restapi/resources/ConnectionsResource.java

```diff
@@ -138,6 +146,61 @@ public Uni<Connection> updateConnection(@PathParam("id") String id, ConnectionSp
         .chain(ignored -> Uni.createFrom().item(() -> getConnectionModel(id)));
   }
 
+  @PATCH
+  @Path("/{id}")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @APIResponses(value = {
+      @APIResponse(
+          responseCode = "200",
+          description = "Connection updated with PATCH",
+          content = {
+              @Content(mediaType = "application/json",
+                  schema = @Schema(implementation = Connection.class))
+          }),
+      @APIResponse(
+          responseCode = "404",
+          description = "Connection not found",
+          content = {
+              @Content(mediaType = "application/json",
+                  schema = @Schema(implementation = Failure.class))
+          }),
+      @APIResponse(
+          responseCode = "401",
+          description = "Could not authenticate",
+          content = {
+              @Content(mediaType = "application/json",
+                  schema = @Schema(implementation = Failure.class))
+          }),
+      @APIResponse(
+          responseCode = "400",
+          description = "Invalid input",
+          content = {
+              @Content(mediaType = "application/json",
+                  schema = @Schema(implementation = Failure.class))
+          }),
+  })
+  public Uni<Connection> patchConnection(
+      @PathParam("id") String id,
+      JsonMergePatch patch
+  ) {
+    ObjectMapper mapper = new ObjectMapper();
+          try {
+            JsonNode existingSpecNode = mapper.valueToTree(connectionStateManager
+                .getConnectionState(id).getSpec());
+
+            JsonNode patchedSpecNode = patch.apply(existingSpecNode);
+            ConnectionSpec patchedSpec = mapper.treeToValue(patchedSpecNode,
+                ConnectionSpec.class);
+            return connectionStateManager
+                .updateSpecForConnectionState(id, patchedSpec)
+                .chain(ignored -> Uni.createFrom().item(() -> getConnectionModel(id)));
+          } catch (JsonPatchException | IOException e) {
+            Log.errorf("Failed to patch connection: {}, Connection ID:{}, Request: {}", e.getMessage(), id, patch);
+            throw new WebApplicationException("Failed to patch connection, please check the format of your request", Response.Status.BAD_REQUEST);
+          }
+        }
```

Comment:
Do you mind reformatting this code and updating the formatted string in line 199? I think using `{}` as placeholders won't work.

```suggestion
    try {
      JsonNode existingSpecNode = mapper.valueToTree(
          connectionStateManager.getConnectionState(id).getSpec());

      JsonNode patchedSpecNode = patch.apply(existingSpecNode);
      ConnectionSpec patchedSpec = mapper.treeToValue(patchedSpecNode, ConnectionSpec.class);
      return connectionStateManager
          .updateSpecForConnectionState(id, patchedSpec)
          .chain(ignored -> Uni.createFrom().item(() -> getConnectionModel(id)));
    } catch (JsonPatchException | IOException e) {
      Log.errorf(
          "Failed to patch connection: %s, Connection ID: %s, Request: %s",
          e.getMessage(),
          id,
          patch
      );
      throw new WebApplicationException(
          "Failed to patch connection, please check the format of your request",
          Response.Status.BAD_REQUEST
      );
    }
  }
```

Reply:
Thanks!